### PR TITLE
Je/upgrade twofactor

### PR DESCRIPTION
**Pull request by emord**
@nickpell 

My changes to the library: https://github.com/Bouke/django-two-factor-auth/pull/178 My branch is based off of an unreleased 1.5.0

### CODE REVIEW

File: settings.py [hunk 1]
```diff
@@ -160,6 +160,7 @@
     'auditcare.middleware.AuditMiddleware',
     'no_exceptions.middleware.NoExceptionsMiddleware',
     'corehq.apps.locations.middleware.LocationAccessMiddleware',
+    'two_factor.middleware.threadlocals.ThreadLocals',
```
Comment (emord) [inline settings.py]: I think that this is technically only needed for the twilio gateway, but just went ahead and added in here.

http://django-two-factor-auth.readthedocs.io/en/stable/configuration.html#twilio-gateway

**Review by nickpell — APPROVED** [review]
🎉

**Comment by nickpell** [comment]
fyi @calellowitz

**Comment by emord** [comment]
Successfully tested this on staging

File: corehq/apps/hqwebapp/two_factor_gateways.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def __init__(self):
     def send_sms(self, device, token):
         message = _('Your authentication token is %s') % token
         self.client.sms.messages.create(
-            to=device.number,
+            to=device.number.as_e164,
```
Comment (emord) [inline corehq/apps/hqwebapp/two_factor_gateways.py]: same format as https://github.com/Bouke/django-two-factor-auth/blob/master/two_factor/gateways/twilio/gateway.py#L56

File: corehq/apps/settings/views.py [hunk 3]
```diff
@@ -232,7 +232,7 @@ def dispatch(self, request, *args, **kwargs):
         if not request.couch_user.is_web_user():
             raise Http404
 
-        return super(BaseMyAccountView, self).dispatch(request, *args, **kwargs)
+        return super(MyProjectsList, self).dispatch(request, *args, **kwargs)
```
Comment (emord) [inline corehq/apps/settings/views.py]: A lot of these super calls were using the incorrect class

File: settings.py [hunk 4]
```diff
@@ -437,7 +437,7 @@
 
 # after login, django redirects to this URL
 # rather than the default 'accounts/profile'
-LOGIN_REDIRECT_URL = '/'
+LOGIN_REDIRECT_URL = 'homepage'
```
Comment (emord) [inline settings.py]: @nickpell I'm wondering if this has something to do with recent url changes. I always got an error locally on redirecting to `/`, but I think `homepage` is the same thing. Does that seem like a recent django change that could have happened?

Comment (nickpell) [inline settings.py]: seems possible - bf88d89b touched this

**Review by nickpell — APPROVED** [review]

**Pull request merged**